src/smtsolvers/LookaheadSMTSolver.cc

```diff
@@ -147,43 +134,46 @@ LookaheadSMTSolver::PathBuildResult LookaheadSMTSolver::setSolverToNode(LANode c
         curr = parent;
         parent = curr->p;
     }
-#ifdef LADEBUG
-    printf("Setting solver to the right dl %d\n", path.size());
-#endif
-    for (int i = path.size() - 1; i >= 0; i--) {
-        newDecisionLevel();
-        if (value(path[i]) == l_Undef) {
-#ifdef LADEBUG
-            printf("I will propagate %s%d\n", sign(path[i]) ? "-" : "", var(path[i]));
-#endif
-            int curr_dl = decisionLevel();
-            uncheckedEnqueue(path[i]);
-            lbool res = laPropagateWrapper();
-            // Here it is possible that the solver is on level 0 and in an inconsistent state.  How can I check this?
-            if (res == l_False) {
-                return PathBuildResult::pathbuild_tlunsat; // Indicate unsatisfiability
-            } else if (res == l_Undef) {
-                cancelUntil(0);
-                return PathBuildResult::pathbuild_restart; // Do a restart
-            }
-            if (curr_dl != decisionLevel()) {
-                return PathBuildResult::pathbuild_unsat;
-            }
+    // setting solver to the correct dl
+    int i = 0;
+    if(path.size() <= decisionLevel()) {
+
+        if (path.size() > 0) { cancelUntil(path.size() - 1);}
+        if (path.size() == 0) { cancelUntil(0);}
+        if(path.size() >= assumptions.size()){
+            crossed_assumptions = assumptions.size();
         } else {
-#ifdef LADEBUG
-            printf("Would propagate %s%d but the literal is already assigned\n", sign(path[i]) ? "-" : "", var(path[i]));
-#endif
-            if (value(path[i]) == l_False) {
-#ifdef LADEBUG
-                printf("Unsatisfiable branch since I'd like to propagate %s%d but %s%d is assigned already\n", sign(path[i]) ? "-" : "", var(path[i]), sign(~path[i]) ? "-" : "", var(path[i]));
-                printf("Marking the subtree false:\n");
-                n->print();
-#endif
-                return PathBuildResult::pathbuild_unsat;
+            crossed_assumptions = path.size();
+        }
+    } else {
+        i = path.size() - decisionLevel() - 1;
```

Comment:
Please add parentheses for clarification.

Reply:
Fixed

Reply:
I meant add parentheses to make it clear if this is `(path.size() - decisionLevel()) - 1` or `path.size() - (decisionLevel() - 1)`